src/core/reducers/abuse.js

```diff
@@ -0,0 +1,87 @@
+/* @flow */
+export const LOAD_ADDON_ABUSE_REPORT = 'LOAD_ADDON_ABUSE_REPORT';
+export const SEND_ADDON_ABUSE_REPORT = 'SEND_ADDON_ABUSE_REPORT';
+
+type LoadAddonAbuseReportType = {
+  addon: {|
+    guid: string,
+    id: number,
+    slug: string,
+  |},
+  message: string,
+  reporter: Object | null,
+};
+
+export function loadAddonAbuseReport(
+  { addon, message, reporter }: LoadAddonAbuseReportType
+) {
+  if (!addon) {
+    throw new Error('addon is required');
+  }
+  if (!message) {
+    throw new Error('message is required');
+  }
+  if (reporter === undefined) {
+    throw new Error('reporter cannot be undefined');
+  }
+
+  return {
+    type: LOAD_ADDON_ABUSE_REPORT,
+    payload: { addon, message, reporter },
+  };
+}
+
+type SendAddonAbuseReportAction = {|
+  addonSlug: string,
+  errorHandlerId: string,
+  message: string,
+|};
+
+export function sendAddonAbuseReport(
+  { addonSlug, errorHandlerId, message }: SendAddonAbuseReportAction
+) {
+  if (!addonSlug) {
+    throw new Error('addonSlug is required');
+  }
+  if (!errorHandlerId) {
+    throw new Error('errorHandlerId is required');
+  }
+  if (!message) {
+    throw new Error('message is required');
+  }
+
+  return {
+    type: SEND_ADDON_ABUSE_REPORT,
+    payload: { addonSlug, errorHandlerId, message },
+  };
+}
+
+export const initialState = {
+  bySlug: {},
+};
+
+type ReducerState = {|
+  bySlug: {
+    [addonSlug: string]: {| message: string, reporter: string |},
+  },
+|};
+
+export default function abuseReducer(
+  state: ReducerState = initialState,
+  action: Object
```

Comment:
btw, the way to type this would be:

```js
action: SendAddonAbuseReportAction
```

but then you also have to type your constants as string literals [like this](https://github.com/mozilla/addons-frontend/blob/master/src/amo/reducers/addonsByAuthors.js#L17) so Flow can figure out which action object is being operated on in each switch case:

```js
export const LOAD_ADDON_ABUSE_REPORT: 'LOAD_ADDON_ABUSE_REPORT' = 'LOAD_ADDON_ABUSE_REPORT';
```

`Object` is ok if you want to leave it as is but you will get some nice safety checks in your reducer code.